src/redux/actions/samples/createSample.js

```diff
@@ -57,30 +56,26 @@ const createSample = (
       },
     );
 
-    const json = await response.json();
-
-    throwIfRequestFailed(response, json, endUserMessages.ERROR_SAVING);
-
     await dispatch({
       type: SAMPLES_CREATE,
       payload: { sample: newSample, experimentId },
     });
 
     return newSampleUuid;
   } catch (e) {
-    const { message } = e;
-    console.error(e);
+    const errorMessage = handleError(e, endUserMessages.ERROR_CREATING_SAMPLE);
 
+    console.log('failed to create sample');
     dispatch({
       type: SAMPLES_ERROR,
       payload: {
-        error: message,
+        error: errorMessage,
       },
     });
 
-    pushNotificationMessage('error', endUserMessages.ERROR_CREATING_SAMPLE);
-
-    throw new Error(message);
+    // throw again the error so `processUpload` won't upload the sample
+    // Promise.reject(errorMessage);
+    throw new Error(errorMessage);
```

Comment:
This seems to be popping up a lot - error handling in Redux. I read more about the recommended way to do error handling in Redux action creators. In both here: https://stackoverflow.com/questions/58266418/correct-way-of-error-handling-in-react-redux and https://exportknowledge.com/article/error-handling-with-react-and-redux people actually use the error field in redux state. 
So when an error happens, all that this file does is to dispatch an "ERROR" action type and it doesn't raise and return anything. Then all components that care about that error state, just check the error in the store. Now I remember that we followed this flow with Data Exploration in the very beginning and it worked very well. I am going to ask someone that knows a lot about front end about this too and then I will tell you what they recommend.

Reply:
agreed, but this one should be done in another refactor